hypertrace-core-graphql-gateway-service-utils/src/main/java/org/hypertrace/core/graphql/utils/gateway/MultiFilterConverter.java

```diff
@@ -0,0 +1,70 @@
+package org.hypertrace.core.graphql.utils.gateway;
+
+import static io.reactivex.rxjava3.core.Single.zip;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.core.Single;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.hypertrace.core.graphql.common.request.AttributeAssociation;
+import org.hypertrace.core.graphql.common.schema.results.arguments.filter.FilterArgument;
+import org.hypertrace.core.graphql.common.utils.Converter;
+import org.hypertrace.gateway.service.v1.common.Filter;
+import org.hypertrace.gateway.service.v1.common.Operator;
+
+public class MultiFilterConverter
+    implements Converter<Collection<Collection<AttributeAssociation<FilterArgument>>>, Filter> {
+
+  private final AttributeExpressionConverter attributeExpressionConverter;
+  private final OperatorConverter operatorConverter;
+  private final LiteralConstantExpressionConverter literalConstantExpressionConverter;
+
+  @Inject
+  MultiFilterConverter(
+      AttributeExpressionConverter attributeExpressionConverter,
+      OperatorConverter operatorConverter,
+      LiteralConstantExpressionConverter literalConstantExpressionConverter) {
+    this.attributeExpressionConverter = attributeExpressionConverter;
+    this.operatorConverter = operatorConverter;
+    this.literalConstantExpressionConverter = literalConstantExpressionConverter;
+  }
+
+  @Override
+  public Single<Filter> convert(
+      Collection<Collection<AttributeAssociation<FilterArgument>>> filters) {
```

Comment:
Why are we assuming a collection of collection of filters should be an `OR-AND` relationship? That's extremely unintuitive. Can we revert this and instead build a real converter here that allows the caller to build relationships - `(collection<filters>, logicalOperator) => filter`